### PR TITLE
Add AWS CloudFormation Scripts for APIM 4.0.0

### LINKED ISSUE #75: Update AWS Cloud Formation scripts for 4.0.0 release
**Description:**
Need to update AWS CF scripts to support the new pattern-1 (APIM + MI).

### PR DESCRIPTION
**Pull request by AyeshW**
## Purpose
This PR:

- modifies AWS CloudFormation scripts
- adds MI resource configurations
- removes analytics resource configurations

### CODE REVIEW

File: product/pattern-1.yaml [hunk 1]
```diff
@@ -529,34 +487,32 @@ Resources:
...
+          ln -s /mnt/efs/server/executionplans /usr/lib/wso2/wso2am/4.0.0/wso2am-4.0.0/repository/deployment/server
 
           if [ ! -d "/mnt/efs/tenants" ]; then
               mkdir -p /mnt/efs/tenants
-              cp -r /usr/lib/wso2/wso2am/3.2.0/wso2am-3.2.0/repository/tenants /mnt/efs
+              cp -r /usr/lib/wso2/wso2am/4.0.0/wso2am-4.0.0/repository/tenants /mnt/efs
           fi
-          rm -rf /usr/lib/wso2/wso2am/3.2.0/wso2am-3.2.0/repository/tenants
-          ln -s /mnt/efs/tenants /usr/lib/wso2/wso2am/3.2.0/wso2am-3.2.0/repository/tenants
+          rm -rf /usr/lib/wso2/wso2am/4.0.0wso2am-4.0.0/repository/tenants
```
Comment (RakhithaRR) [inline product/pattern-1.yaml]: 4.0.0/wso2am-4.0.0

**Review by pubudu538 — APPROVED** [review]

File: product/pattern-1.yaml (hunk 1 above)
Comment (AyeshW) [reply product/pattern-1.yaml]: Noted

File: product/pattern-1.yaml [hunk 2]
```diff
@@ -940,31 +906,30 @@ Resources:
...
       HealthCheckTimeoutSeconds: 5
       HealthyThresholdCount: 4
-      HealthCheckPath: /analytics-dashboard
-      HealthCheckPort: 9643
+      HealthCheckPort: 8253
       Matcher:
         HttpCode: 200
-      Name: apim-carbon-9643
-      Port: 9643
+      Name: mi-8243
```
Comment (RakhithaRR) [inline product/pattern-1.yaml]: should be `mi-8253` right?

File: product/pattern-1.yaml [hunk 3]
```diff
@@ -1513,7 +1272,7 @@ Parameters:
     Description: VPC ID of your existing Virtual Private Cloud (VPC)
   WSO2APIMPrivateSubnet1:
     Type: AWS::EC2::Subnet::Id
-    Description: Private Subnet for PuppetMaster, Prodcut Instance, EFS and the DB
+    Description: Private Subnet for PuppetMaster, Product Instance, EFS and the DB
```
Comment (RakhithaRR) [inline product/pattern-1.yaml]: Let's remove the EFS part

**Review by RakhithaRR — APPROVED** [review]

**Comment by RakhithaRR** [comment]
Fixes https://github.com/wso2/aws-apim/issues/75

**Pull request merged**